### ISSUE TITLE
Crash under Mac OS X

### ISSUE THREAD

**Issue by mkoppanen** [Nuitka/Nuitka#389]
Mac OS X, using python from homebrew and nuitka version 0.6.3.1. 
$ python --version
Python 3.7.3

The problem in seems to be in MainProgram.c in the argument conversion. The minimal way to reproduce the crash is to compile the following program with nuitka:

```
import sys
print(len(sys.argv))
```
$ ./t.bin 
Segmentation fault: 11

```
(lldb) bt 
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x603040)
  * frame #0: 0x00007fff5bd206da libsystem_c.dylib`wcslen + 8
    frame #1: 0x000000010017075d Python`PyUnicode_FromWideChar + 72
    frame #2: 0x00000001001ed354 Python`PySys_SetArgvEx + 108
    frame #3: 0x0000000100004585 t.bin`main + 469
    frame #4: 0x00007fff5bc78ed9 libdyld.dylib`start + 1
```

Changing MainProgram.c from (line 156 in convertCommandLineParameters):

```
#ifdef __APPLE__
        argv_copy[i] = _Py_DecodeUTF8_surrogateescape(argv[i], strlen(argv[i]));
#elif PYTHON_VERSION < 350
        argv_copy[i] = _Py_char2wchar(argv[i], NULL);
#else
        argv_copy[i] = Py_DecodeLocale(argv[i], NULL);
#endif
```
to

```
#if PYTHON_VERSION >= 350
		argv_copy[i] = Py_DecodeLocale(argv[i], NULL);
#elif defined(__APPLE__) && PYTHON_VERSION >= 320
		argv_copy[i] = _Py_DecodeUTF8_surrogateescape(argv[i], strlen(argv[i]));
#else
		argv_copy[i] = _Py_char2wchar(argv[i], NULL);      
#endif
```
fixes the issue for me and the program runs without errors. The reference from modifications was found here: https://bitbucket.org/mpi4py/mpi4py/issues/21/new-release-of-any-kind

**Comment by kayhayen** [comment]
Thanks a lot, want to make it a PR? Otherwise, just your name+email would be fine then I would attribute it to you.

Yours,
Kay

**Comment by mkoppanen** [comment]
@kayhayen,

I'll open a PR later today with the relevant changes. 

Thanks,
Mikko

**Comment by mkoppanen** [comment]
The PR is here: https://github.com/Nuitka/Nuitka/pull/391

@kayhayen Thanks for your for your work on such a great tool.

**Comment by kayhayen** [comment]
Released as 0.6.4
